nengo/utils/tests/test_socket.py

```diff
@@ -0,0 +1,151 @@
+import math
+import threading
+import numpy as np
+
+import nengo
+from nengo.utils import sockets
+
+
+class SimThread(threading.Thread):
```

Comment:
There is no shared state to protect. The threads are using the lock to wait for a threading event signal. This is used in the `test_send_recv_chain` test where it is necessary for the threads to be started in a certain order (the sender thread needs to start sending information before the receiving threads start running).